pandas/core/resample.py

```diff
@@ -1689,55 +1755,56 @@ def _get_period_range_edges(first, last, offset, closed="left", base=0):
     if not all(isinstance(obj, Period) for obj in [first, last]):
         raise TypeError("'first' and 'last' must be instances of type Period")
 
-    # GH 23882
+    # GH 23882 & 31809
     first = first.to_timestamp()
     last = last.to_timestamp()
-    adjust_first = not offset.is_on_offset(first)
-    adjust_last = offset.is_on_offset(last)
+    adjust_first = not freq.is_on_offset(first)
+    adjust_last = freq.is_on_offset(last)
 
     first, last = _get_timestamp_range_edges(
-        first, last, offset, closed=closed, base=base
+        first, last, freq, closed=closed, origin=origin, offset=offset,
     )
 
-    first = (first + adjust_first * offset).to_period(offset)
-    last = (last - adjust_last * offset).to_period(offset)
+    first = (first + adjust_first * freq).to_period(freq)
+    last = (last - adjust_last * freq).to_period(freq)
     return first, last
 
 
-def _adjust_dates_anchored(first, last, offset, closed="right", base=0):
+def _adjust_dates_anchored(first, last, freq, closed="right", origin=None, offset=None):
     # First and last offsets should be calculated from the start day to fix an
     # error cause by resampling across multiple days when a one day period is
-    # not a multiple of the frequency.
-    #
-    # See https://github.com/pandas-dev/pandas/issues/8683
```

Comment:
do you have tests from this issue?

Reply:
Yes, I just changed the comment to remove the link of the issue into GH 8683.
The test is untouched on my branch:

https://github.com/pandas-dev/pandas/blob/60353d5652d89bff9f2a71d05452bafe986f4c4c/pandas/tests/resample/test_datetime_index.py#L938

Reply:
That the same one from https://github.com/pandas-dev/pandas/pull/8941/files, but it seems that it has been moved (not by me) since 2014.